apps/client/src/hooks.server.ts

```diff
@@ -1,35 +1,54 @@
-
-import { SvelteKitAuth } from "@auth/sveltekit";
-import GitHub from "@auth/core/providers/github";
-import Google from "@auth/core/providers/google";
+import { SvelteKitAuth } from '@auth/sveltekit';
+import GitHub from '@auth/core/providers/github';
+import Google from '@auth/core/providers/google';
 import { sequence } from '@sveltejs/kit/hooks';
 import { redirect } from '@sveltejs/kit';
-import type { Handle } from "@sveltejs/kit";
+import type { Handle } from '@sveltejs/kit';
 
+interface User {
+  name: string | null;
+  image: string | null;
+  created_at: string | null;
+  role: string | null;
+  updated_at: string | null;
+  username: string | null;
+  hasConnectedGithub: boolean;
+  githubUsername: string | null;
```

Comment:
I think we can auto-generate these from our GraphQL schema. Codegen could be of help here.

Reply:
We could then augment the generated types with some [utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype).